toolkit/tools/pkg/imagecustomizerlib/imageutils.go

```diff
@@ -278,3 +286,29 @@ func createPartIdToPartUuidMap(partIDToDevPathMap map[string]string, diskPartiti
 
 	return partIdToPartUuid, nil
 }
+
+func extractOSRelease(rootfsPartition *diskutils.PartitionInfo, buildDir string) (string, error) {
+	tmpDir := filepath.Join(buildDir, "tmp_rootfs")
+
+	// Temporarily mount the rootfs partition
+	rootfsPartitionMount, err := safemount.NewMount(rootfsPartition.Path, tmpDir, rootfsPartition.FileSystemType, 0, "", true)
+	if err != nil {
+		return "", fmt.Errorf("failed to mount rootfs partition (%s):\n%w", rootfsPartition.Path, err)
+	}
+	defer rootfsPartitionMount.Close()
+
+	// Read /etc/os-release
+	osReleasePath := filepath.Join(tmpDir, "etc/os-release")
+	data, err := os.ReadFile(osReleasePath)
```

Comment:
nit: Use `file.Read`.

It's basically the same thing but includes some extra logging which can be nice.